syntax = "proto3";

package brain.action.v1;

import "brain/shared/v1/envelope.proto";

service SwitchboardService {
  rpc Health(SwitchboardHealthRequest) returns (SwitchboardHealthResponse);
}

message SwitchboardHealthStatus {
  bool service_ready = 1;
  bool adapter_ready = 2;
  bool cas_ready = 3;
  string detail = 4;
}

message SwitchboardHealthPayload {}

message SwitchboardHealthRequest {
  brain.shared.v1.EnvelopeMeta metadata = 1;
  SwitchboardHealthPayload payload = 2;
}

message SwitchboardHealthResponse {
  brain.shared.v1.EnvelopeMeta metadata       = 1;
  SwitchboardHealthStatus payload             = 2;
  repeated brain.shared.v1.ErrorDetail errors = 3;
}
